nuxeo-retention-web/elements/nuxeo-retention-analytics.js

```diff
@@ -0,0 +1,379 @@
+      return entry.key;
+    });
+  }
+
+  _extendEndDate(date) {
+    if (date && moment) {
+      return this._formatDate(moment(date).add(1, 'days').subtract(1, 'ms').toJSON());
+    }
+    return date;
+  }
+
+  _formatDateLabels(data, format) {
+    return data.map((entry) => {
+      if (Array.isArray(entry.value)) {
+        return entry.value.map(this._formatDateLabels);
+      }
+      return moment(entry.key).format(format);
+    });
+  }
+
+  _formatLabels(data, prefix) {
+    return data.map((entry) => {
+      if (Array.isArray(entry.value)) {
+        entry.value.map(this._formatLabels);
+      }
+      return this.i18n((prefix && prefix + entry.key) || entry.key);
+    });
+  }
+
+  _chartOptions() {
+    return {
+      legend: {
+        display: true,
+        position: 'right',
+        labels: {
+          boxWidth: 25,
+          fontSize: 12,
+        },
+      },
+      animation: {
+        duration: 100,
+        easing: 'easeOutQuart',
+        onComplete: this._drawValuesInPie,
+      },
+    };
+  }
+
+  // Source: https://stackoverflow.com/questions/33363373/how-to-display-pie-chart-data-values-of-each-slice-in-chart-js
+  // (to draw values inside the pie charts)
+  _drawValuesInPie() {
+    const { ctx } = this.chart;
+    ctx.textAlign = 'center';
+    ctx.textBaseline = 'bottom';
+    ctx.fillStyle = '#fff';
+
+    this.data.datasets.forEach((dataset) => {
+      for (let i = 0; i < dataset.data.length; i += 1) {
+        const model = dataset._meta[Object.keys(dataset._meta)[0]].data[i]._model;
+        const { total } = dataset._meta[Object.keys(dataset._meta)[0]];
+        const { startAngle, endAngle } = model;
+        const midAngle = startAngle + (endAngle - startAngle) / 2;
+        const midRadius = model.innerRadius + (model.outerRadius - model.innerRadius) / 2;
+        const x = midRadius * Math.cos(midAngle);
+        const y = midRadius * Math.sin(midAngle);
+
+        const percent = Math.round((dataset.data[i] / total) * 100);
+        const percentStr = `${percent} %`;
+        // Don't Display If Legend is hide or value is 0
+        const dataSetMeta = dataset._meta;
+        const dataSetMetaFirstProp = dataSetMeta[0] || dataSetMeta[Object.keys(dataSetMeta)[0]];
+
+        if (dataset.data[i] !== 0 && dataSetMetaFirstProp && !dataSetMetaFirstProp.data[i].hidden) {
+          if (percent > 5) {
+            ctx.fillText(dataset.data[i], model.x + x, model.y + y);
+            // Display percent in another line, line break doesn't work for fillText
+            ctx.fillText(percentStr, model.x + x, model.y + y + 15);
+          }
+        }
+      }
+    });
+  }
+
+  _formatDateNow() {
+    return moment().format('YYYY-MM-DD');
+  }
+
+  _formatDateFutureInMonths(count) {
+    return moment().add(count, 'months').format('YYYY-MM-DD');
+  }
+
+  // Returns YYYY-MM instead of YYYY-MM-DD
+  _futureExpirationLabels(data) {
+    return this._labels(data).map((label) => label.substring(0, 7));
```

Comment:
If we want only Year and Month, we can use the `YYYY-MM` format directly like:

`moment().format('YYYY-MM');`

Reply:
thanks for the suggestion, done.